assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -11700,7 +11700,7 @@ public String getGrade(boolean overrideWithGradebookValue)
 				String gradeGB=this.getGradeForUserInGradeBook(null);
 				if(gradeGB!=null)
 				{
-					rv=gradeGB;
+					rv=!gradeGB.equals("")?gradeGB:null;
```

Comment:
Ternaries are complex, especially negative ones. Could you please expand this to make maintenance easier?